skbio/metadata/_interval.py

```diff
@@ -0,0 +1,343 @@
+            self._is_stale_tree = False
+
+        invs = set()
+
+        # Find queries by interval
+        if intervals is not None:
+            for value in intervals:
+                invs.update(self._query_interval(value))
+
+        # Find queries by feature attribute
+        if len(invs) == 0 and metadata is not None:
+            invs = self._metadata
+
+        if metadata is not None:
+            invs = self._query_attribute(invs, metadata)
+        return list(invs)
+
+    def drop(self, intervals=None, boundaries=None, metadata=None):
+        """ Drops Interval objects according to a specified query.
+
+        Drops all Interval objects that matches the query.
+
+        Parameters
+        ----------
+        intervals : iterable of tuple of ints
+            A list of intervals associated with the Interval object.
+        boundaries : iterable of tuple of bool
+            A list of boundaries associated with the Interval object.
+        metadata : dict
+            A dictionary of key word attributes associated with the
+            Interval object.
+
+        Examples
+        --------
+        >>> from skbio.metadata import IntervalMetadata
+        >>> interval_metadata = IntervalMetadata()
+        >>> interval_metadata.add(intervals=[(0, 2), (4, 7)],
+        ...                       boundaries=None, metadata={'name': 'sagA'})
+        >>> interval_metadata.add(intervals=[(40, 70)],
+        ...                       boundaries=None, metadata={'name': 'sagB'})
+        >>> interval_metadata.drop(metadata={'name': 'sagA'})
+        >>> interval_metadata.query(metadata={'name': 'sagA'})
+        []
+        """
+        if intervals is None:
+            intervals = []
+        if metadata is None:
+            metadata = {}
+
+        queried_invs = self.query(intervals=intervals,
+                                  boundaries=boundaries,
+                                  metadata=metadata)
+        new_invs = []
+        # iterate through queries and drop them
+        for inv in self._metadata:
+            if inv not in queried_invs:
+                new_invs.append(inv)
+        self._metadata = new_invs
+        self._is_stale_tree = True
+
+    def __eq__(self, other):
+        return sorted(self._metadata) == sorted(other._metadata)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __repr__(self):
+        if len(self._metadata) < 10:
+            return str(self._metadata)
+        else:
+            return str(self._metadata[:5] + ['...'] + self._metadata[-5:])
+
+
+def _assert_valid_interval(interval):
+    if isinstance(interval, tuple):
+        if len(interval) == 2:
+            start, end = interval
+            if start > end:
+                raise ValueError("`start` is greater than `end`.")
+        else:
+            raise ValueError("An interval must be a tuple of exactly "
+                             "two coordinates, not %r" % repr(interval))
```

Comment:
The reason why this was a problem, was because `interval` is a tuple, which breaks this string formatting.

The following is a fix.  Hope that is ok
```python
            raise ValueError("An interval must be a tuple of exactly "
                             "two coordinates, not %r" % (interval, ))

```